spyderlib/widgets/varexp/arrayeditor.py

```diff
@@ -444,6 +474,14 @@ def _sel_to_text(self, cell_range):
         if not cell_range:
             return
         row_min, row_max, col_min, col_max = get_idx_rect(cell_range)
+        if col_min == 0 and col_max == (self.model().cols_loaded-1):
+            # we've selected a whole column. It isn't possible to
+            # select only the first part of a column without loading more, 
+            # so we can treat it as intentional and copy the whole thing
+            col_max = self.model().total_cols-1
+        if row_min == 0 and row_max == (self.model().rows_loaded-1):
+            row_max = self.model().total_rows-1

```

Comment:
I really like this, this is **very** clever and solves the select columns, rows and all content in a very elegant way!